metrics/web/test/dashboard/presentation/state/project_metrics_notifier_test.dart

```diff
@@ -188,77 +217,115 @@ void main() {
 
-      final firstProjectMetrics =
-          projectMetricsNotifier.projectsMetricsTileViewModels.first;
+        final projectMetrics =
+            projectMetricsNotifier.projectsMetricsTileViewModels.first;
+        final buildNumberMetric = projectMetrics.buildNumberMetric;
+        final actualNumberOfBuilds = buildNumberMetric.numberOfBuilds;
 
-      expect(
-        firstProjectMetrics.buildNumberMetric.numberOfBuilds,
-        expectedBuildNumberMetrics.numberOfBuilds,
-      );
-    });
+        expect(actualNumberOfBuilds, equals(expectedNumberOfBuilds));
+      },
+    );
 
-    test("loads the performance metrics", () async {
-      final period = ReceiveProjectMetricsUpdates.buildsLoadingPeriod.inDays;
-      final expectedPerformanceMetrics =
-          expectedProjectMetrics.performanceMetrics;
+    test(
+      "loads the correct number number of performance points from the build day project metrics",
+      () async {
+        final period = ReceiveProjectMetricsUpdates.buildsLoadingPeriod.inDays;
+        final expectedNumberOfPoints = period + 1;
 
-      final buildPerformance =
-          expectedPerformanceMetrics.buildsPerformance.first;
-      final expectedX = List.generate(period + 1, (index) => index);
-      final expectedY = List.generate(period, (_) => 0);
-      expectedY.add(buildPerformance.duration.inMilliseconds);
+        final firstProjectMetrics =
+            projectMetricsNotifier.projectsMetricsTileViewModels.first;
+        final performanceMetrics = firstProjectMetrics.performanceSparkline;
+        final performancePoints = performanceMetrics.performance;
 
-      final firstProjectMetrics =
-          projectMetricsNotifier.projectsMetricsTileViewModels.first;
-      final performanceMetrics = firstProjectMetrics.performanceSparkline;
-      final performancePoints = performanceMetrics.performance;
+        expect(performancePoints, hasLength(expectedNumberOfPoints));
+      },
+    );
 
-      expect(performancePoints, hasLength(equals(period + 1)));
-      expect(
-        performanceMetrics.value,
-        equals(expectedPerformanceMetrics.averageBuildDuration),
-      );
-      expect(performancePoints.map((p) => p.x), equals(expectedX));
-      expect(performancePoints.map((p) => p.y), equals(expectedY));
-    });
+    test(
+      "loads the performance metrics from the build day project metrics with the correct average duration",
+      () async {
+        final performance = expectedBuildDayProjectMetrics.performanceMetric;
+        final expectedDuration = performance.averageBuildDuration;
+
+        final firstProjectMetrics =
+            projectMetricsNotifier.projectsMetricsTileViewModels.first;
+        final performanceMetrics = firstProjectMetrics.performanceSparkline;
+
+        expect(performanceMetrics.value, equals(expectedDuration));
+      },
+    );
+
+    test(
+      "loads the performance metrics from the build day project metrics",
```

Comment:
Would it be great to test that the notifier creates performance points with a Y axis equal to `0` if there are no builds performed during the day in the period?